tripal/tripal
# Fix the sequence coordinates field

# Bug Fix

### Closes #1859

### Tripal Version: 4.x

## Description
The sequence coordinates field was preventing us from editing a gene.
The reason is that the "linking table" for this field is the `featureloc` table. It contains a foreign key to the `feature` table i.e. `feature_id`, it contains another key back to the feature table. This is because a feature has a location on another feature, to put it briefly.

But this so-called "linking table" also has numerous other columns, such as `fmin`, `fmax`, `strand`, etc.
For chado storage to work, we need all of these columns defined in the field properties, and passed through in the widget. 

Making this harder was that some of the columns did not have terms defined. This part was fixed in PR #1816

This PR adds the missing properties, and includes an update function to fix this on an existing site.

## Testing?

 1. Import the type collection "Genomic Content Types"
 2. Create an organism [Saccharomyces cerevisiae](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=559292)
 3. Create an analysis
 4. Import this example file
[issue1824.gff3.txt](https://github.com/tripal/tripal/files/15109905/issue1824.gff3.txt)
For default landmark type, I used `DNA (SO:0000352)`
 5. Run the job `drush trp-run-jobs --username=drupaladmin --root=/var/www/drupal/web`
 6. Publish the "Gene" content type
 7. Run the job `drush trp-run-jobs --username=drupaladmin --root=/var/www/drupal/web`
 8. View the published "PAU8" gene. Observe the sequence location as `NC_001133.9:1806..2169-`
 9. Edit the gene, don't make any changes, and save.
 10. Verify that there are no errors, and that the sequence location is unchanged `NC_001133.9:1806..2169-`

It would also be good to build a 4.x docker and repeat steps 1 through 5, then switch to this branch, and run `drush updatedb` to run the update function before continuing
